src/components/Signup.jsx

```diff
@@ -33,20 +49,66 @@ const Signup = () => {
   const [showMessage, setShowMessage] = useState(false);
   const [loading, setLoading] = useState(false);
 
-  const handleSubmit = async () => {
+  const handleSubmit = async (e) => {
     setLoading(true);
-    if (password === rePassword) {
+    if (password === rePassword && validatePhone(phone)) {
       const response = await signUp({name, email, phone, password});
       setLoading(false);
-      if (response.ok) {
-        navigate('/');
+      if (response) {
+        if (response.name) {
+          const loginResponse = await login({email, password});
+          if (loginResponse.jwt) {
```

Comment:
this is getting very hard to read. Try to avoid nested if statements.

Can you try and refactor this to first validate the fields, and only then make the `signup` request? 

Perhaps a separate function to `validateFields`? 

And try do it without nesting `if` statements

Going forward, I highly recommend looking into third-party tools for forms and validation. They will save you a lot of boiler plate and improve readability.

Personally, I'm a big fan of Formik and yup together.

Reply:
I will check them out